app/src/components/TodoList.vue

```diff
@@ -0,0 +1,194 @@
+<template>
+  <div>
+    <div class="inputs-container">
+      <div class="input-group mb-3 create-container">
+        <div class="input-group-prepend">
+          <button @click="addItem(title)" type="button" class="btn btn-success" id="button-addon1"><img src="../assets/add.svg" alt="trash" width="20"></button>
+        </div>
+        <input type="text" class="form-control" placeholder="" aria-describedby="button-addon1" v-model="title">
+      </div>
+      <div class="input-group flex-nowrap filter">
+        <div class="input-group-prepend">
+          <button class="btn btn-dark" id="addon-wrapping">
+            <img src="../assets/filter.svg" alt="filter" width="20">
+          </button>
+        </div>
+        <input type="text" class="form-control" placeholder="Find todo" @change="getFilteredList(filter)" v-model="filter">
+      </div>
+    </div>
+    <ul class="list-container">
+      <li v-for="(item, index) in paginatedData" :key="item.id" @click="selectItem(index)" class="list-group-item">
+        <span>{{item.title}}</span>
+        <div v-show="selectedIndex===index" class="btn-container">
+          <button @click="removeItem(item)" type="button" class="btn btn-danger"><img src="../assets/trash.svg" alt="trash" width="20"></button>
+          <div class="input-group mb-3 input-container">
+            <input type="text" class="form-control" placeholder="Recipient's username" aria-label="Recipient's username" aria-describedby="button-addon2" v-model="item.title">
+            <div class="input-group-append">
+              <button @click="updateItem(item)" class="btn btn-warning" type="button" id="button-addon2"><img src="../assets/edit.svg" alt="edit" width="20"></button>
+            </div>
+          </div>
+        </div>
+      </li>
+    </ul>
+    <nav>
+      <ul class="pagination">
+        <li class="page-item">
+          <button
+            class="page-link"
+            :disabled="pageNumber === 0" 
+            @click="prevPage"
+          >
+            Previous
+          </button>
+        </li>
+        <div class="page-link page-number">{{pageNumber + 1}}</div>
+        <li class="page-item">
+          <button 
+            class="page-link" 
+            :disabled="pageNumber >= pageCount -1" 
+            @click="nextPage">
+            Next
+          </button>
+        </li>
+      </ul>
+    </nav>
+  </div>
+</template>
+
+<script>
+const API_URL = `http://localhost:3000/todos`;
+export default {
+  name: "todoList",
+  data() {
+    return {
+      items: [],
+      selectedIndex: null,
+      updatingIndex: null,
+      title:'',
+      pageNumber: 0,
+      paginatedData: [],
+      pageCount: 0,
+      filter: ''
+    }
+  },
+
+  methods: {
+    getFilteredList(filter){
```

Comment:
filter can be extracted into a method, no need to pass it as an argument
Computed prop is usually used for this, not method